src/machines/java/com/enderio/machines/common/blockentity/base/MachineBlockEntity.java

```diff
@@ -457,4 +476,35 @@ public RedstoneControl getRedstoneControl() {
     public void setRedstoneControl(RedstoneControl redstoneControl) {
         this.redstoneControl = redstoneControl;
     }
+
+    //only called on server
+    @Override
+    public InteractionResult onWrenched(UseOnContext context) {
+        Player player = context.getPlayer();
+        if(player != null && level!= null && player.isSecondaryUseActive()) {//aka break block
+            BlockPos pos = context.getClickedPos();
+            BlockState state = context.getLevel().getBlockState(pos);
+            List<ItemStack> drops = Block.getDrops(state, (ServerLevel) level,pos, level.getBlockEntity(pos));//level is always ServerLevel since method only called on server.
+            level.setBlock(pos, Blocks.AIR.defaultBlockState(), Block.UPDATE_ALL_IMMEDIATE);
+            player.swing(InteractionHand.MAIN_HAND);
+            //TODO: custom sound when sound manager is up and running??
+            SoundType soundType = state.getBlock().getSoundType(state,level,pos,null);
+            level.playSound(null, pos,soundType.getBreakSound(), SoundSource.BLOCKS,soundType.volume, soundType.pitch);
+            for (ItemStack drop: drops) {
+                if(!player.addItem(drop)){
+                    level.addFreshEntity(new ItemEntity(level, pos.getX()+0.5, pos.getY()+0.5, pos.getZ()+0.5, drop));
+                }
+            }
```

Comment:
Sure, added it in a class called PlayerInteractionUtil in the core source folder since I didn't find any other good place for it. If you have another place in mind, let me know!